### PR TITLE
ReversePackedSegsOp optimized GPU code

### PR DESCRIPTION
**Pull request by slayton58**
Removes the need for all the Copy calls, in one of our apps reduced time from ~40ms to < 200us

### CODE REVIEW

File: caffe2/operators/reverse_packed_segs_op.cu [hunk 1]
```diff
@@ -2,6 +2,91 @@
...
+template <typename T, typename LengthType>
+__global__
+void ReversePackedSegments_kernel(
+      size_t max_length,
+      size_t batch_size,
+      size_t block_size,
+      const LengthType* lengths_ptr,
+      const T* data_ptr,
+      T* rev_data_ptr) {
+
```
Comment (pietern) [inline caffe2/operators/reverse_packed_segs_op.cu]: Mega nit, trailing whitespace. Internal linter trips over this. Could you configure your editor to automatically strip this?

Comment (slayton58) [reply caffe2/operators/reverse_packed_segs_op.cu]: Sure, I'll look into that (I've had it setup on vim before) - I'll update the PR in a few minutes

**Review by akyrola — APPROVED** [review]
Looks good to me, assuming you run the unit test for it:

caffe2/python/operator_test/sequence_ops_test.py

File: caffe2/operators/reverse_packed_segs_op.cu (hunk 1 above)
Comment (slayton58) [reply caffe2/operators/reverse_packed_segs_op.cu]: Took longer than expected, but updated

**Comment by slayton58** [comment]
And yes, I ran this through the unit test

**Pull request closed without merging**